docker/drone/src/test/java/org/arquillian/cube/docker/drone/VncRecorderLifecycleManagerTest.java

```diff
@@ -0,0 +1,165 @@
+                CubeDroneConfiguration.fromMap(conf),
+                seleniumContainers
+                );
+
+        assertThat(new File(destination, "org_arquillian_cube_docker_drone_VncRecorderLifecycleManagerTest_shouldMoveRecordingVideo.flv").exists(), is(true));
+    }
+
+    @Test
+    public void shouldStopVncAfterEachTestByDefault() throws IOException, NoSuchMethodException {
+
+        final File destination = temporaryFolder.newFolder("destination");
+        final File video = temporaryFolder.newFile("file.flv");
+
+        when(seleniumContainers.getVideoRecordingFile()).thenReturn(video.toPath());
+        when(after.getTestClass()).thenReturn(new TestClass(VncRecorderLifecycleManagerTest.class));
+        when(after.getTestMethod()).thenReturn(VncRecorderLifecycleManagerTest.class.getMethod("shouldMoveRecordingVideo"));
+
+        Map<String, String> conf = new HashMap<>();
+        conf.put("videoOutput", destination.getAbsolutePath());
+
+        TestResult testResult = TestResult.passed();
+
+        VncRecorderLifecycleManager vncRecorderLifecycleManager = new VncRecorderLifecycleManager();
+        vncRecorderLifecycleManager.vnc = cube;
+        vncRecorderLifecycleManager.stopRecording(after,
+                testResult,
+                CubeDroneConfiguration.fromMap(conf),
+                seleniumContainers
+        );
+
+        verify(cube).stop();
+        verify(cube).destroy();
+    }
+
+    @Test
+    public void shouldDiscardRecordingIfConfiguredInOnlyFailingAndPassedTest() throws IOException, NoSuchMethodException {
+
+        final File destination = temporaryFolder.newFolder("destination");
+        final File video = temporaryFolder.newFile("file.flv");
+
+        when(seleniumContainers.getVideoRecordingFile()).thenReturn(video.toPath());
+        when(after.getTestClass()).thenReturn(new TestClass(VncRecorderLifecycleManagerTest.class));
+        when(after.getTestMethod()).thenReturn(VncRecorderLifecycleManagerTest.class.getMethod("shouldDiscardRecordingIfConfiguredInOnlyFailingAndPassedTest"));
+
+        Map<String, String> conf = new HashMap<>();
+        conf.put("videoOutput", destination.getAbsolutePath());
+        conf.put("recordingMode", "ONLY_FAILING");
+
+        TestResult testResult = TestResult.passed();
+
+        VncRecorderLifecycleManager vncRecorderLifecycleManager = new VncRecorderLifecycleManager();
+        vncRecorderLifecycleManager.vnc = cube;
+        vncRecorderLifecycleManager.stopRecording(after,
+                testResult,
+                CubeDroneConfiguration.fromMap(conf),
+                seleniumContainers
+        );
+
+        assertThat(new File(destination, "org_arquillian_cube_docker_drone_VncRecorderLifecycleManagerTest_shouldDiscardRecordingIfConfiguredInOnlyFailingAndPassedTest.flv").exists(), is(false));
+    }
+
+    @Test
+    public void shouldMoveRecordingIfConfiguredInOnlyFailingAndFailedTest() throws IOException, NoSuchMethodException {
```

Comment:
Somehow this 'should_move_recording_if_configured_in_only_failing_and_failed_test' reads better for my taste.

Shall we try to follow this convention?

In addition it will make it easier to find a test method in the stack trace :)